### ISSUE TITLE
kiwix-serve failing to start on 400GB RPi 4 until ~15min after boot (fails on 1GB RAM model, works on 2GB model)

### ISSUE THREAD

**Issue by holta** [iiab/iiab#1905]
In this example below, I finally got kiwix-serve (i.e. the systemd service) to start 17min after boot.

This seems to be the pattern -- kiwix-serve was continuously unworkable (unable to start, due to its timeout) for the full 10-15min after boot, each time I tried that on the 1GB RAM RPi 4,.

Note that kiwix-serve grabbed 271.4 MB when it finally started (as seen below) which seems quite high :(?)

Is it possible this very high memory consumption might be a consequence of having so many (about 43) ZIM files as mentioned at https://github.com/iiab/iiab/issues/1866#issuecomment-515688222 ?  Or a particular subset of troublesome ZIM files, from the list seen at https://paste.fedoraproject.org/paste/~M~rAIjEVc6Jjvhk7dHPBw -> https://paste.ubuntu.com/p/gJ5WTwkyFQ/ ?

(Certainly there should be error cleaner messaging to warn others, as soon a we figure this out-)

```
root@box:~# systemctl status kiwix-serve.service
● kiwix-serve.service - Provides the Kiwix ZIM File Server
   Loaded: loaded (/etc/systemd/system/kiwix-serve.service; enabled; vendor pres
   Active: active (running) since Sat 2019-07-27 02:05:12 EAT; 30s ago
  Process: 4707 ExecStart=/opt/iiab/kiwix/bin/kiwix-serve --daemon --port 3000 -
 Main PID: 4771 (kiwix-serve)
    Tasks: 5 (limit: 1599)
   Memory: 271.4M
   CGroup: /system.slice/kiwix-serve.service
           └─4771 /opt/iiab/kiwix/bin/kiwix-serve --daemon --port 3000 --nolibra

Jul 27 02:04:18 box.lan systemd[1]: Starting Provides the Kiwix ZIM File Server.
Jul 27 02:05:12 box.lan systemd[1]: Started Provides the Kiwix ZIM File Server.

root@box:~# uptime
 02:05:53 up 17 min,  3 users,  load average: 1.94, 2.73, 2.33
```

**Comment by holta** [comment]
Prior to [IIAB 7.0](https://github.com/iiab/iiab/wiki/IIAB-7.0-Release-Notes) release, I hope to investigate if PRs #1913 & #1914 make a material difference, by simulating such conditions with 40+ ZIM files on a 1GB RPi 4 especially.

Of course it's possible these 3 kiwix PR's (added since last week) might make a slight difference:
#1909 #1913 #1914

**Comment by holta** [comment]
This is confirmed solved, apparently as a result of PRs #1913 [`TimeoutStartSec=180` in kiwix-serve.service] and its companion PR #1914.  Possibly #1909 [kiwix 2.1.0] had a small effect too?  In any case, when applying these 3 PR's, the problem entirely goes away and...
- `systemctl status kiwix-serve` shows "Active (running)" within 2 minutes of booting!
- `htop` shows that memory consumption on the 1GB RAM RPi 4 briefly rises just over 500MB between 1min and 2min after booting...but then within a minute of that, falls to around 300MB total memory consumption.
